src/pymovements/gaze/eyetracker.py

```diff
@@ -0,0 +1,99 @@
+# Copyright (c) 2022-2023 The pymovements Project Authors
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+"""This module holds the EyeTracker class."""
+from __future__ import annotations
+
+from pymovements.utils import checks
+from pymovements.utils import decorators
+
+
+@decorators.auto_str
+class EyeTracker:
```

Comment:
can you use the `@dataclass` decorator here instead?
as this class won't have any methods it would be best to implement this class as a dataclass right from the start.

You won't need to define an `__init__()` method then, but you will need to move the checks to a `__post_init__()` method. The docstring example can be moved to the class docstring.

See this page with info about dataclasses:
https://docs.python.org/3/library/dataclasses.html

And a nice video on why dataclasses are awesome:
https://www.youtube.com/watch?v=CvQ7e6yUtnw

Reply:
I've made this change